test/hotspot/jtreg/compiler/loopopts/parallel_iv/TestParallelIvInIntCountedLoop.java

```diff
@@ -0,0 +1,250 @@
+/*
+ * Copyright (c) 2024 Red Hat and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package compiler.loopopts.parallel_iv;
+
+import compiler.lib.ir_framework.Argument;
+import compiler.lib.ir_framework.Arguments;
+import compiler.lib.ir_framework.IR;
+import compiler.lib.ir_framework.IRNode;
+import compiler.lib.ir_framework.Test;
+import compiler.lib.ir_framework.TestFramework;
+
+import java.util.function.Function;
+
+/**
+ * @test
+ * @bug 8328528
+ * @summary test the long typed parallel iv replacing transformation for int counted loop
+ * @library /test/lib /
+ * @run main compiler.loopopts.parallel_iv.TestParallelIvInIntCountedLoop
+ */
+public class TestParallelIvInIntCountedLoop {
+    public static void main(String[] args) {
+        TestFramework.run();
+        testCorrectness();
+    }
+
+    /*
+     * The IR framework can only test against static code, and the transformation relies on strides being constants to
+     * perform constant propagation. Therefore, we have no choice but repeating the same test case multiple times with
+     * different numbers.
```

Comment:
That is not exactly true. You could create a `static final` field that gets its value assigned randomly. This would then be a constant once the code is JIT-ed. Just an idea if you want to expand your testing with some random numbers.